### PR TITLE
keygen.js: Initial implementation

### PR DESCRIPTION
**Pull request by CyberShadow**

### CODE REVIEW

File: keygen.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
+/* jshint esversion : 8 */
```
Comment (CyberShadow) [inline keygen.js]: Needs license

File: keygen.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+
+	async function genSpkac(algorithm, pkeyopts) {
+		if (algorithm === undefined)
+			algorithm = 'RSA';
+		if (!algorithm.length || algorithm.match(/[\s]/))
+			throw 'Invalid algorithm';
+		pkeyopts = pkeyopts.split(/[\s]{1,}/g).filter(Boolean);
+
+		await openSSL.runCommand(
+			"genpkey" +
```
Comment (nomis) [inline keygen.js:160]: RSA defaults to 1024 bits, this should specify 4096 bits

Comment (CyberShadow) [reply keygen.js:160]: MDN says:

> For RSA keys, the keyparams parameter is not used (ignored if present). The user may be given a choice of RSA key strengths. Currently, the user is given a choice between "high" strength (2048 bits) and "medium" strength (1024 bits).

0. We should pick up `keytype` (and maybe other properties) from the tag
1. We can add a `keysize` parameter which translates to `rsa_keygen_bits:XXX`, or allow setting `keyopts` on the tag which would get directly concatenated to the value in the `keyopts` input field. (The input field can be hidden with CSS if the site doesn't want users to customize options.)

Comment (nomis) [reply keygen.js:160]: Well, for my purposes I don't want people to choose and only want 4096 bits

Comment (CyberShadow) [reply keygen.js:160]: Yep, we can do that but as this library aims to be a generic polyfill it probably should stick as close to the behaviour of `<keygen>` as possible. We can customize for our needs on site.

File: keygen.js (hunk 1 above)
Comment (CyberShadow) [reply keygen.js]: Done

Comment (CyberShadow) [reply keygen.js]: Using the same license as openpgp.js (Apache 2.0).

File: keygen.js (hunk 2 above)
Comment (CyberShadow) [reply keygen.js:160]: > We should pick up `keytype` (and maybe other properties) from the tag

Done

> We can add a `keysize` parameter which translates to `rsa_keygen_bits:XXX`, or allow setting `keyopts` on the tag which would get directly concatenated to the value in the `keyopts` input field.

Added `-keygenjs-pkeyopts` with the same semantics as the value of the options input (and `-pkeyopts` OpenSSL switch).

File: keygen.js [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+		if (disabled)
+			optInput.setAttribute('disabled', '');
+		keygen.appendChild(optInput);
+
+		var status = document.createElement('span');
+		status.setAttribute('class', 'keygen-status');
+		keygen.appendChild(status);
+
+		var resultLink = document.createElement('a');
+		status.setAttribute('class', 'keygen-link');
```
Comment (nomis) [inline keygen.js]: Should be `resultLink`, not `status`.

**Pull request merged**